src/h5web/visualizations/shared/utils.ts

```diff
@@ -112,3 +112,15 @@ export function getTicksProp(
     ? { tickValues: getIntegerTicks(visibleDomain, numTicks) }
     : { numTicks };
 }
+
+export function assertNumOrStr(val: unknown): asserts val is number | string {
+  if (typeof val !== 'number' && typeof val !== 'string') {
+    throw new Error('Expected number or string');
+  }
+}
+
+export function assertArray<T>(val: unknown): asserts val is T[] {
+  if (!Array.isArray(val)) {
+    throw new Error('Expected array');
+  }
+}
```

Comment:
I wonder if we can go further than that by writing custom types for 1D and 2D ndarrays and a typeguard that will check the dimensionnality of the ndarray.
What do you think ? Perhaps in another PR ?

Reply:
Hmm, I did wonder a bit about this, but decided it wasn't worth it. `MappedVis` doesn't really need stronger typing on its `value` prop since we just flatten the array right at the start. What matters here is telling TypeScript that the `unknown` value is just an `array` of some sort.

If the value array's dimensions don't match the dataset's metadata dimensions, it's an error in the file, so we don't really care what happens. If the vis breaks, good; if it doesn't, good. Throwing ourselves isn't worth it, since there could be other problems, like values in the array that are not numbers.